packages/dcos-integration-test/extra/test_metrics.py

```diff
@@ -44,30 +44,35 @@ def test_metrics_masters_prom(dcos_api_session):
         assert response.status_code == 200, 'Status code: {}'.format(response.status_code)
 
 
-@retrying.retry(wait_fixed=2000, stop_max_delay=150 * 1000)
-def get_metrics_prom(dcos_api_session, node, expected_metrics):
-    """Assert that expected metrics are present on prometheus port on node.
+@retrying.retry(wait_fixed=5000, stop_max_delay=300 * 1000)
+def check_metrics_prom(dcos_api_session, node, check_func):
+    """Get metrics from prometheus port on node and run check_func function,
+    asserting that it returns True.
 
-    Retries on non-200 status or missing expected metrics
-    for up to 150 seconds.
+    Retries on non-200 status or failed assertions
+    for up to 300 seconds.
 
     """
-    response = dcos_api_session.session.request('GET', 'http://{}:61091/metrics'.format(node))
+    response = dcos_api_session.session.request(
+        'GET', 'http://{}:61091/metrics'.format(node))
     assert response.status_code == 200, 'Status code: {}'.format(response.status_code)
-    for metric_name in expected_metrics:
-        assert metric_name in response.text
+    assert check_func(response) is True
```

Comment:
Let's leave this `get_metrics_prom()`, have it return the response, and let the test execute its assertions, rather than passing in a callable that the helper executes.